### PR TITLE
 reviews

### PR DESCRIPTION
**Pull request by BouaikIlham**
- [ ] Redux: Fetch data and update the Redux store
Upon first rendering fetch data from the SpaceX API endpoints:

   - Rockets: https://api.spacexdata.com/v3/rockets
   - Missions: https://api.spacexdata.com/v3/missions
 - [ ] the data are fetched, dispatch an action to store the selected data in Redux store:

     - Rockets:
                      id
                      rocket_name
                      description
                      flickr_images
     - Missions:
                       mission_id
                       mission_name
                       description
- [ ] Render UI:lists
- [ ] Written actions and reducers for booking rockets  missions
- [ ] Written actions and reducers for canceling rockets and leaving missions
- [ ] conditional components rendering
- [ ] Compose two/three-column layout and list ONLY the rockets reserved and missions joined by the user (as per design):
       - Render a list of all joined missions (use filter()).
       - Render a list of all reserved rockets (use filter()).
- [ ] added test for all  components and functions.
- [ ] Added linters
- [ ] Opened a pull request from the development branch to the main branch with all changes made by me.

### CODE REVIEW

File: src/components/Mission.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          </button>
+        ) : (
+          <button
+            type="button"
+            className="join-btn"
+            id={id}
+            onClick={() => dispatch(joinMission(id))}
+          >
+            Join mission
+          </button>
```
Comment (Omar-Muhamad) [inline src/components/Mission.js:40]: - [ ] Nice job using React conditional renering.

File: src/components/Rocket.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            >
+              Cancel Reservation
+            </button>
+          </div>
+        ) : (
+          <div>
+            <p>{description}</p>
+            <button type="button" onClick={() => dispatch(reserveRocket(id))}>
+              Reserve Rocket
+            </button>
```
Comment (Omar-Muhamad) [inline src/components/Rocket.js:39]: - [ ] Same here Nice job using React conditional renering.

File: src/Redux/Missions/missions.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      description: item.description,
+      member: false,
+    };
+    missions.push(mission);
+  });
+  dispatch({
+    type: 'loadMissions',
+    playload: missions,
+  });
+};
```
Comment (Omar-Muhamad) [inline src/Redux/Missions/missions.js:47]: - [ ] [Optional] For a more professional approach all async actions put them in a separate .js module you can name it something like apiHelper.js and put it directly in the Redux directory.

File: src/Redux/Rockets/rockets.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      description: item.description,
+      flickr_images: [...item.flickr_images],
+    };
+    rockets.push(rocket);
+  });
+  dispatch({
+    type: LOADROCKETS,
+    playload: rockets,
+  });
+};
```
Comment (Omar-Muhamad) [inline src/Redux/Rockets/rockets.js:51]: - [ ] [Optional] Same here, For a more professional approach all async actions put them in a separate .js module you can name it something like apiHelper.js and put it directly in the Redux directory.

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @BouaikIlham,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### ⚠️Please consider the Optional suggestions below⚠️



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- Please check the inline comments below ⬇.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**